### PR TITLE
Fix seeking with PTS values exceeding max int

### PR DESCRIPTION
**Pull request by rbuehlma**
This is a follow up to https://github.com/xbmc/xbmc/pull/12086

I found some additional places where double got pushed into an int.

### CODE REVIEW

File: xbmc/cores/VideoPlayer/Edl.cpp [hunk 1]
```diff
@@ -770,16 +770,16 @@ int CEdl::RemoveCutTime(int iSeek) const
   return iSeek - iCutTime;
 }
 
-int CEdl::RestoreCutTime(int iClock) const
+double CEdl::RestoreCutTime(double iClock) const
```
Comment (ksooo) [inline xbmc/cores/VideoPlayer/Edl.cpp]: If you change the data type, then please also change the variable's name. Prefix "i" stands for "int". Applies also to other places of your commit.

**Review by ksooo — CHANGES_REQUESTED** [review]

File: xbmc/cores/VideoPlayer/Edl.cpp (hunk 1 above)
Comment (rbuehlma) [inline xbmc/cores/VideoPlayer/Edl.cpp]: Sure, here we are :)

File: xbmc/cores/VideoPlayer/Edl.cpp [hunk 2]
```diff
@@ -770,19 +770,19 @@ int CEdl::RemoveCutTime(int iSeek) const
...
+    return dClock;
 
-  int iSeek = iClock;
+  double dSeek = dClock;
   for (int i = 0; i < (int)m_vecCuts.size(); i++)
   {
-    if (m_vecCuts[i].action == CUT && iSeek >= m_vecCuts[i].start)
-      iSeek += m_vecCuts[i].end - m_vecCuts[i].start;
+    if (m_vecCuts[i].action == CUT && dSeek >= m_vecCuts[i].start)
+      dSeek += (double)m_vecCuts[i].end - (double)m_vecCuts[i].start;
```
Comment (ksooo) [inline xbmc/cores/VideoPlayer/Edl.cpp]: please use c++ style casts

**Review by ksooo — CHANGES_REQUESTED** [review]

File: xbmc/cores/VideoPlayer/Edl.cpp (hunk 2 above)
Comment (rbuehlma) [inline xbmc/cores/VideoPlayer/Edl.cpp]: I removed the cast. If I understand C++ correctly, the calculation is done in double any way as it expands to dSeek = dSeek + int.
As we calculate here with a double, the int is promoted to a double.
Can you confirm that?

Comment (garbear) [inline xbmc/cores/VideoPlayer/Edl.cpp]: Better include the cast. Windows warns about this conversion:

```
ActiveAE.cpp(1911): warning C4244: 'initializing': conversion from 'int64_t' to 'double', possible loss of data
```

**Review by FernetMenta — APPROVED** [review]

File: xbmc/cores/VideoPlayer/Edl.cpp (hunk 2 above)
Comment (rbuehlma) [inline xbmc/cores/VideoPlayer/Edl.cpp]: Done. Better now?

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Review by ksooo — APPROVED** [review]

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Pull request merged**